### PR TITLE
check poison records after decryption for mysql [T685]

### PR DESCRIPTION
**Pull request by Lagovas**
* move detecting poison records after failed decryption of acrastruct
* add fast check on block size to prevent extra decryption poison records with zone mode on

### CODE REVIEW

File: decryptor/postgresql/pg_decryptor.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func handlePoisonCheckResult(decryptor base.Decryptor, poisoned bool, err error)
 // checkInlinePoisonRecordInBlock check block on poison record as whole AcraStruct block (only when IsPoisonRecordCheckOn() == true)
 func checkInlinePoisonRecordInBlock(block []byte, decryptor base.Decryptor, logger *log.Entry) error {
 	// check is it Poison Record
-	if decryptor.IsPoisonRecordCheckOn() {
+	if decryptor.IsPoisonRecordCheckOn() && len(block) > base.GetMinAcraStructLength() {
```
Comment (vixentael) [inline decryptor/postgresql/pg_decryptor.go]: why not `>=`?

Comment (Lagovas) [reply decryptor/postgresql/pg_decryptor.go]: too lazy to type one more symbol

Comment (vixentael) [reply decryptor/postgresql/pg_decryptor.go]: 😂

**Comment by vixentael** [comment]
Great!

Note for future us:
This PR supplements #230

**Pull request merged**